src/nyaml/nyaml2nxdl.py

```diff
@@ -811,6 +820,54 @@ def verbose_flag(verbose, keyword, value):
         print(f"key:{keyword}; value type is {type(value)}\n")
 
 
+def xml_handle_nametype(keyword, keyword_name, dct, obj):
+    """
+    Identify NeXus nameType attribute for field, group, attribute use hint if required.
+    """
+    concept_name = keyword_name[2:] if keyword_name.startswith("\\@") else keyword_name
+    if concept_name == "":
+        # no explicit name given as e.g. in group type="NXobject"
+        # obj.set("nameType", "any")
+        # cannot be specified because having no/empty concept name is not allowed
+        # cannot be partial because if no hint is given why should it be partial
+        return
+    if concept_name.islower():
+        # obj.set("nameType", "specified")  # is the NeXus default, no need to add
+        return
+    if concept_name.isupper():
+        # nameType="any" correct for almost all cases except for those where an
+        # explicit nameType hint is made in the yaml file
+        # this is relevant for e.g. NXcanSAS ENTRY/DATA/Q which is specified
+        if dct.get(keyword) and "nameType" in dct[keyword]:
+            supported = ["specified", "any", "partial"]
+            name_type = dct[keyword]["nameType"]
+            if name_type in supported:
+                obj.set("nameType", name_type)
+            else:
+                raise ValueError(f"nameType for {keyword} is not in {supported}")
+        else:
+            obj.set("nameType", "any")
```

Comment:
If nameType was not declared in nyaml, we shall not write anything to nxdl.xml either. We can check and give warnings if we see some potentially unintentional inconsistencies, like a declared `nameType=any` (or `partial`) with no uppercase letters in the name, but we shall not stop, autocorrect, or extend the declarations beyond the defined content.

Reply:
@mkuehbach agree! I need to implement that here!

Reply:
warnings e.g.  partial but no UPPER

Reply:
Addressed

Reply:
done